File: src/codility/java/greedy/TieRopes.java
package codility.java.greedy;

public class TieRopes {

    public int solution(int K, int[] A) {
        int sum = 0;
        int ans = 0;
        for (int a: A) {
            sum += a;
            if (sum >=K) {
                ans++;
                sum = 0;
            }
        }


        return ans;
    }
}

File: src/codility/kotlin/TieRopes.kt
package codility.kotlin

class TieRopes {

    fun solution(K: Int, A: IntArray): Int {
        var count = 0
        var current = 0L

        for ( a in A) {
            current += a
            if (current >= K) {
                current = 0
                count++
            }
        }
        return count
    }

}